viite-UI/src/view/RoadNamingToolWindow.js

```diff
@@ -97,10 +98,21 @@
             $('.modal-overlay').remove();
         }
 
+      var getDataObjects = function (fieldValue) {
+        var fieldDateString = fieldValue.split('.');
+        var fieldDate = new Date(fieldDateString[2], fieldDateString[1], fieldDateString[0]);
+        var futureDate = new Date();
+        futureDate.setFullYear(futureDate.getFullYear() + yearLimit);
+        return {'fieldDate': fieldDate, 'futureDate': futureDate};
+      };
+
         function toggleSaveButton() {
             $('#saveChangedRoads').prop("disabled",
                 !_.every($('input.form-control[data-fieldname="roadName"],input.form-control[data-fieldname="startDate"]'), function (element) {
-                    return $(element).val() !== '';
+                  if ($(element).attr('data-FieldName') === "startDate") {
+                    var datas = getDataObjects($(element).val());
```

Comment:
"datas" is not a english word. Word data is used as plural. So the right grammar would be "data" instead of "datas".